infobip-openapi-mcp-core/src/main/java/com/infobip/openapi/mcp/auth/web/InitialAuthenticationFilter.java

```diff
@@ -75,44 +72,50 @@ public class InitialAuthenticationFilter extends OncePerRequestFilter {
 
     private final RestClient restClient;
     private final AuthProperties authProperties;
-    private final Optional<OAuthProperties> oAuthProperties;
-    private final OpenApiMcpProperties openApiMcpProperties;
     private final ErrorModelWriter errorModelWriter;
     private final ApiRequestEnricherChain enricherChain;
     private final McpRequestContextFactory contextFactory;
-    private final Optional<ScopeDiscoveryService> scopeDiscoveryService;
-    private final XForwardedHostCalculator xForwardedHostCalculator;
+    private final Optional<WwwAuthenticateProvider> wwwAuthenticateProvider;
+    private final Optional<JwtScopeService> jwtScopeService;
 
     public InitialAuthenticationFilter(
             RestClient restClient,
             AuthProperties authProperties,
-            Optional<OAuthProperties> oAuthProperties,
-            OpenApiMcpProperties openApiMcpProperties,
             ErrorModelWriter errorModelWriter,
             ApiRequestEnricherChain enricherChain,
             McpRequestContextFactory contextFactory,
-            Optional<ScopeDiscoveryService> scopeDiscoveryService,
-            XForwardedHostCalculator xForwardedHostCalculator) {
+            Optional<WwwAuthenticateProvider> wwwAuthenticateProvider,
+            Optional<JwtScopeService> jwtScopeService) {
         this.authProperties = authProperties;
         this.restClient = restClient;
-        this.oAuthProperties = oAuthProperties;
-        this.openApiMcpProperties = openApiMcpProperties;
         this.errorModelWriter = errorModelWriter;
         this.enricherChain = enricherChain;
         this.contextFactory = contextFactory;
-        this.scopeDiscoveryService = scopeDiscoveryService;
-        this.xForwardedHostCalculator = xForwardedHostCalculator;
+        this.wwwAuthenticateProvider = wwwAuthenticateProvider;
+        this.jwtScopeService = jwtScopeService;
     }
 
     @Override
     protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain filterChain)
             throws ServletException, IOException {
         var authHeader = request.getHeader(HttpHeaders.AUTHORIZATION);
         if (authHeader == null) {
-            writeUnauthorizedResponse(request, response);
+            writeWwwAuthenticateResponse(request, response, HttpStatus.UNAUTHORIZED);
             return;
         }
 
+        if (jwtScopeService.isPresent()) {
+            var jwtService = jwtScopeService.get();
+            var tokenClaims = jwtService.decodeJwtToken(authHeader);
+            if (tokenClaims != null) {
```

Comment:
This series of if-checks gets simpler after we move all the logic into the `jwtScopeService`. Then this filter can "just" call one verify method and if it returns false we set www auth response header. I'm a little bit concerned with too many `ifPresent` and `!= null` checks, it's easy to miss some configuration case and have a broken config <-> impl.